lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -562,6 +562,24 @@ defmodule Kernel.WarningTest do
     purge(Sample)
   end
 
+  test "expands args and warn for atoms beginning with `Elixir.`" do
+    message = ~r"invalid value for option :as, expected an alias, got: :\"Elixir.foobar\""
+
+    assert capture_err(fn ->
+             Code.eval_string("""
+               alias :lists, as: :"Elixir.foobar"
+             """)
+           end) =~ message
+
+    message = ~r"invalid value for option :as, expected an alias, got: Foobar"
+
+    assert capture_err(fn ->
+             Code.eval_string("""
+               alias :lists, as: :"Elixir.Foobar"
```

Comment:
It is actually fine for it to be an atom. Sometimes you may generate an alias name dynamically and use `unquote(generated)` when building the alias and that would break it, since `generated` is most likely an atom.

My suggestion is to keep today's code as is but check that, right after we split on ".", than the first letter of the alias is in upcase.

Reply:
I still think that is a bit strange because the following is confusing:
```
iex(1)> alias :lists, as: :"Elixir.Foobar"
:lists
iex(2)> :"Elixir.Foobar"
Foobar
iex(3)> Foobar          
:lists
```
It feels that the best thing would be to have some kind of `Macro.alias/1` like there is `Macro.var/2`, but I imagine that it will break too many libraries.

Reply:
I agree it is weird if you implement it by hand but most likely that would be generated through a macro or Macro.concat and forcing the user to step back to an alias would be a bit awkward and a backwards incompatible change. :)